fabric/Makefile

```diff
@@ -9,20 +9,27 @@ FABRIC_VERSION ?= 2.1.0
 FABRIC_VERSION_TAG =v$(FABRIC_VERSION)
 FABRIC_PEER_BIN = $(FABRIC_PATH)/build/bin/peer
 FABRIC_TLCC = $(FABRIC_PATH)/core/scc/tlcc
+TLCC_GO_SRC = ${TOP}/tlcc/*.go
+TLCC_ENCLAVE_DIR = ${TOP}/tlcc_enclave
+TLCC_ENCLAVE_LIB = ${TLCC_ENCLAVE_DIR}/_build/libtl.so
 
-FABRIC_CURRENT_TAG = $(shell cd $(FABRIC_PATH); git tag --points-at HEAD)
 FABRIC_PATCHED = $(FABRIC_PATH)/.fpc_patched
-FABRIC_TRACKED_CHANGES = $(shell cd $(FABRIC_PATH); git status -s | grep -v '\?')
 FABRIC_CURRENT_DIFF_COMMITS_CMD = git log $(FABRIC_VERSION_TAG)..HEAD --oneline
 
 build: native
 
 patch: $(FABRIC_PATCHED)
 
-$(FABRIC_PATCHED):
+$(FABRIC_PATCHED): *.patch $(FABRIC_PATH)/.git
+	# Note: .git dependency hopefully captures significant changes in fabric dir requiring repatch (and rebuild)
 	@echo "Patching Fabric ..."; \
-	if [ "$(FABRIC_CURRENT_TAG)" != "$(FABRIC_VERSION_TAG)" ]; then \
-		echo "Aborting! Tag on current HEAD ($(FABRIC_CURRENT_TAG)) does not match expected tag $(FABRIC_VERSION_TAG)!"; \
+	if [ -f $(FABRIC_PATCHED) ]; then \
+		echo "Need to re-apply patch, so try to undo patches first"; \
+		$(MAKE) clean-patch; \
+	fi; \
+	FABRIC_CURRENT_TAG=$$(cd $(FABRIC_PATH); git tag --points-at HEAD); \
+	if [ "$${FABRIC_CURRENT_TAG}" != "$(FABRIC_VERSION_TAG)" ]; then \
+		echo "Aborting! Tag on current HEAD ($${FABRIC_CURRENT_TAG}) does not match expected tag $(FABRIC_VERSION_TAG)!"; \
```

Comment:
$${FABRIC_CURRENT_TAG} is not displayed :(

It looks like that ... `Aborting! Tag on current HEAD () does not match expected tag v2.1.0!`

Reply:
can you be a bit more specific? what's the state of your fabric dir, where did your run it (presumably in dev container)? what is the output before hand? What do you get if you do  manually `(FABRIC_PATH); git tag --points-at HEAD)`?

Reply:
how I tested it ... checkout out your PR code and started into the dev container by `cd utils/docker; make run`. Inside the container I `cd $FABRIC_PATH; git checkout master`

When I run `git tag --points-at HEAD)` there it shows ...
```
root@docker-desktop:/project/src/github.com/hyperledger/fabric# git tag --points-at HEAD
root@docker-desktop:/project/src/github.com/hyperledger/fabric#
```

changing back to fpc folder:
```
root@docker-desktop:/project/src/github.com/hyperledger/fabric# cd $FPC_PATH
root@docker-desktop:/project/src/github.com/hyperledger-labs/fabric-private-chaincode#
root@docker-desktop:/project/src/github.com/hyperledger-labs/fabric-private-chaincode#
root@docker-desktop:/project/src/github.com/hyperledger-labs/fabric-private-chaincode# cd fabric/
root@docker-desktop:/project/src/github.com/hyperledger-labs/fabric-private-chaincode/fabric# make
# Note: .git dependency hopefully captures significant changes in fabric dir requiring repatch (and rebuild)
Patching Fabric ...
Aborting! Tag on current HEAD () does not match expected tag v2.1.0!
Makefile:24: recipe for target '/project/src/github.com/hyperledger/fabric/.fpc_patched' failed
make: *** [/project/src/github.com/hyperledger/fabric/.fpc_patched] Error 1
```

Doing this I can see that it works as expected ... but I thought we should maybe display the commit id the head points to.

Reply:
ok no need to change this here in this PR